modules/indexer/issues/db/db.go

```diff
@@ -49,28 +49,33 @@ func (i *Indexer) Search(ctx context.Context, options *internal.SearchOptions) (
 	//        But the two functions are used in modules/notification/indexer, that means we will import services/indexer in modules/notification/indexer.
 	//        So that's the root problem:
 	//        The notification is defined in modules, but it's using lots of things should be in services.
-
 	cond := builder.NewCond()
-
 	if options.Keyword != "" {
 		repoCond := builder.In("repo_id", options.RepoIDs)
 		if len(options.RepoIDs) == 1 {
 			repoCond = builder.Eq{"repo_id": options.RepoIDs[0]}
 		}
-		subQuery := builder.Select("id").From("issue").Where(repoCond)
 
-		cond = builder.Or(
-			db.BuildCaseInsensitiveLike("issue.name", options.Keyword),
-			db.BuildCaseInsensitiveLike("issue.content", options.Keyword),
-			builder.In("issue.id", builder.Select("issue_id").
-				From("comment").
-				Where(builder.And(
-					builder.Eq{"type": issue_model.CommentTypeComment},
-					builder.In("issue_id", subQuery),
-					db.BuildCaseInsensitiveLike("content", options.Keyword),
-				)),
-			),
-		)
+		if options.IsKeywordNumeric() {
+			cond = builder.And(
+				builder.Eq{"`index`": options.Keyword},
+				repoCond,
+			)
+		} else {
```

Comment:
I wouldn't put it in an `else`.
The `if` is fine, but why not search for issues containing this number as well?

Reply:
I do not understand. I will search for all issues with that same index and from the list of repos matched by the repoCond, which in fact are those repos to which the user has access to.

Reply:
That's not my point.
I mean that when you search for a number, it can actually be intended that you want to see issues containing this number.
For example, say you found the duration of a slow operation in an issue and copied it, but don't know which issue it was exactly.

Reply:
Ah, I see. I will change that.

Reply:
I also found out that the repository condition will be applies by the all to `issue_model.Issues(...)`, so this can be removed as well.